geode-core/src/main/resources/org/apache/geode/internal/sanctioned-geode-core-serializables.txt

```diff
@@ -470,7 +470,7 @@ org/apache/geode/management/internal/cli/domain/RegionAttributesInfo,true,336184
 org/apache/geode/management/internal/cli/domain/RegionDescription,true,6461449275798378332,cndEvictionAttributes:java/util/Map,cndPartitionAttributes:java/util/Map,cndRegionAttributes:java/util/Map,dataPolicy:org/apache/geode/cache/DataPolicy,isAccessor:boolean,isLocal:boolean,isPartition:boolean,isPersistent:boolean,isReplicate:boolean,name:java/lang/String,regionDescPerMemberMap:java/util/Map,scope:org/apache/geode/cache/Scope
 org/apache/geode/management/internal/cli/domain/RegionDescriptionPerMember,true,1,hostingMember:java/lang/String,isAccessor:boolean,name:java/lang/String,regionAttributesInfo:org/apache/geode/management/internal/cli/domain/RegionAttributesInfo,size:int
 org/apache/geode/management/internal/cli/domain/RegionInformation,true,1,dataPolicy:org/apache/geode/cache/DataPolicy,isRoot:boolean,name:java/lang/String,parentRegion:java/lang/String,path:java/lang/String,scope:org/apache/geode/cache/Scope,subRegionInformationSet:java/util/Set
-org/apache/geode/management/internal/cli/domain/StackTracesPerMember,true,1,memberNameOrId:java/lang/String,stackTraces:byte[]
+org/apache/geode/management/internal/cli/domain/StackTracesPerMember,true,1,memberNameOrId:java/lang/String,stackTraces:byte[],timestamp:java/time/Instant
```

Comment:
Would `StackTracesPerMember` ever be deserialized from an older version that didn't include the timestamp field? I'm just asking — I don't actually know. Maybe @jujoramos would know. If this could happen, then the timestamp field would be null and the `formatter.format(...)` line in `ExportStackTraceCommand` could throw a NPE.

Reply:
@aaronlindsey: good catch, I haven't thought about it but my guess is that it could happen, yes. I'm not a 100% sure, though.

Reply:
Do we need to use `try/catch` on formatter or override `toData/fromData` in `StackTracesPerMember`?

Reply:
I would do something like the following: Add package private method in `ExportStackTraceCommand` which handles generating the header message. Then you can write a unit test for this more easily. I don't see an easy way to write an integeration (D-unit) test for this case.

```java
  @VisibleForTesting
  String getHeaderMessage(StackTracesPerMember stackTracesPerMember) {
    String headerMessage = stackTracesPerMember.getMemberNameOrId();

    Instant timestamp = stackTracesPerMember.getTimestamp();
    if (timestamp != null) {
      headerMessage += " at " + formatter.format(timestamp);
    }

    return headerMessage;
  }
``` 

The issue with overriding `readObject` in `StackTracesPerMember` is that you wouldn't know which value to assign to the `timestamp` field when the object is deserialized.